api/src/routing.js

```diff
@@ -139,6 +139,17 @@ app.use(prometheusMiddleware({
     45, 90, 180, 360, 600,
     1200, 1800, 3600
   ],
+  additionalLabels: ['userRoles'],
+  extractAdditionalLabelValuesFn: (req) => {
+    const userRoles = req.userCtx?.roles || [];
+    const presentableRoles = [...userRoles]
+      .sort()
+      .join(',');
+    
+    return {
+      userRoles: presentableRoles,
```

Comment:
I really hate that `prometheus-api-metrics` does not support emitting multiple time series (with different label values) per request.  😞 My understanding of Prometheus "best practices" would be that we should just have a `user_role` label and then emit multiple time series when a user has multiple roles (one for each role).  This would make it simple to slice and dice the metrics by role.  As things are here, we will have to do some regexing in the queries...

Unfortunately, I spent a bunch of time trying alternatives here and none of them were satisfactory.  I think what you have here is the best we can do (so no changes needed).